src/middlewares/errorHandler.js

```diff
@@ -0,0 +1,6 @@
+const handleErrors = (err, req, res, next) => {
+  //  Add logger when it's merged
+  res.status(500).json({ message: 'Server ran into an error', err });
+};
+
+module.exports = handleErrors;
```

Comment:
Will this middleware use any params or run any conditions to check there's an error?

Reply:
My intention was build an error handler that we can use instead of express' built-in error handler. The benefit we get from doing that is that we can log our error. After checking the docs again, I'm adding a new condition to check if the server started writing response. Please let me know if I miss something

Reply:
Okay now that you've added the condition to check if response headers were sent, it makes sense. Earlier I was wondering when will the error handler take effect.

Just 1 small suggestion you can rename this function to `errorHandler` which is the name you've used when importing the function in app.js file. Or you could use this approach without any naming:

```js
module.exports = (err, req, res, next) => {
.
.
.
.
```